packages/url.lua

```diff
@@ -32,25 +125,55 @@ SILE.registerCommand("href", function (options, content)
       borderoffset = options.borderoffset },
       content)
   else
+    -- FIXME
+    -- Calling \href{someurl} is not documented in the description below
+    -- but surely for a good reason, as pdf:link currently puts the content in
+    -- an hbox, and our linebreaks won't be applied...
     options.src = content[1]
-    local breakpat = options.breakpat or "/"
-    content = inputfilter.transformContent(content, urlFilter, breakpat)
     SILE.call("pdf:link", { dest = options.src, external = true,
       borderwidth = options.borderwidth,
       borderstyle = options.borderstyle,
       bordercolor = options.bordercolor,
       borderoffset = options.borderoffset },
       function (_, _)
-        SILE.call("code", {}, content)
+        SILE.call("url", { language = options.language }, content)
       end)
   end
-end)
+end, "Inserts a PDF hyperlinks.")
 
 SILE.registerCommand("url", function (options, content)
-  local breakpat = options.breakpat or "/"
-  local result = inputfilter.transformContent(content, urlFilter, breakpat)
-  SILE.call("code", {}, result)
-end)
+  SILE.settings.temporarily(function ()
+    local primaryPenalty = SILE.settings.get("url.linebreak.primaryPenalty")
+    local secondaryPenalty = SILE.settings.get("url.linebreak.secondaryPenalty")
+    local worsePenalty = primaryPenalty + secondaryPenalty
+
+    if options.language then
+      SILE.languageSupport.loadLanguage(options.language)
+      if options.language == "fr" then
+        -- Trick the engine by declaring a "fake"" language that doesn't apply
+        -- the typographic rules for punctuations
+        SILE.hyphenator.languages["_fr_noSpacingRules"] = SILE.hyphenator.languages.fr
+        -- Not needed (the engine already defaults to SILE.nodeMakers.unicode if
+        -- the language is not found):
+        -- SILE.nodeMakers._fr_noSpacingRules = SILE.nodeMakers.unicode
+        SILE.settings.set("document.language", "_fr_noSpacingRules")
+      else
+        SILE.settings.set("document.language", options.language)
+      end
+    else
+      SILE.settings.set("document.language", 'und')
+    end
+
+    local result = inputfilter.transformContent(content, urlFilter, {
+      primaryPenalty = primaryPenalty,
+      secondaryPenalty = secondaryPenalty,
+      worsePenalty = worsePenalty
+    })
+    SILE.call("code", {}, result) -- important: do not pass the options through
```

Comment:
Would there be any practical use for passing the options through? We can always clear whatever font/language property is an issue and pass everything else through — but only if there is any conceivable use case for that being a net benefit.

Reply:
Comment removed, ok.